### PR TITLE
[ADD] Connector Equipment

### PR DESCRIPTION
**Pull request by osi-scampbell**
This PR adds connector_equipment module.

#25 

Note:
Unable to log chatter in service profile from this module because this module does not depend on agreement_serviceprofile. Chatter will be logged from connector_equipment_service instead.

### CODE REVIEW

File: connector_equipment/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    'author': 'Open Source Integrators, Odoo Community Association (OCA)',
+    "website": 'https://github.com/OCA/vertical-isp',
+    'depends': ['maintenance'],
+    'data': [
+        'views/maintenance_equipment.xml',
+        'views/backend_equipment.xml',
+        'security/ir.model.access.csv'
+    ],
+    'installable': True,
+    'auto_install': True,
```
Comment (dreispt) [inline connector_equipment/__manifest__.py]: No point in auto-installing.

**Review by wolfhall — APPROVED** [review]

File: connector_equipment/models/backend_equipment.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+from odoo import fields, models
+
+
+class MaintenanceEquipment(models.Model):
+    _name = 'backend.equipment'
+    _description = 'Backend Equipment'
+
+    name = fields.Char('Name')
+    managed = fields.Boolean('Can be managed')
```
Comment (max3903) [inline connector_equipment/models/backend_equipment.py]: @osi-scampbell This field should remain on `maintenance.equipment`.

File: connector_equipment/models/backend_equipment.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    host = fields.Char('Host')
+    port = fields.Char('Port')
+    user = fields.Char('User')
+    password = fields.Char('Password')
+    protocol = fields.Selection([('http_json', 'HTTP/JSON'),
+                                 ('rest', 'REST')], 'Protocol')
+
+    maintenance_equipment_id = fields.Many2one('maintenance.equipment',
+                                               string="Maintenance Equipment")
```
Comment (max3903) [inline connector_equipment/models/backend_equipment.py]:
```
equipment_ids = fields.One2many('maintenance.equipment', 'backend_id',
                                string="Equipments")
```

File: connector_equipment/models/maintenance_equipment.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class MaintenanceEquipment(models.Model):
+    _inherit = 'maintenance.equipment'
+    _description = 'Maintenance Equipment'
+
+    backend_equipment_id = fields.Many2one('backend.equipment',
```
Comment (max3903) [inline connector_equipment/models/maintenance_equipment.py]:
```suggestion
    backend_id = fields.Many2one('backend.equipment',
```

File: connector_equipment/readme/USAGE.rst [hunk 5]
```diff
@@ -0,0 +1,4 @@
+On a maintenance.equipment, fill out the necessary information to connect
+to your API. Overwrite the methods in maintenance.equipment however you want
+to connect to your API.
+
```
Comment (max3903) [inline connector_equipment/readme/USAGE.rst]:
```suggestion
```

File: connector_equipment/views/backend_equipment.xml [hunk 6]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <!-- Branch Tree View -->
```
Comment (max3903) [inline connector_equipment/views/backend_equipment.xml]:
```suggestion
    <!-- Backend Tree View -->
```

File: connector_equipment/views/backend_equipment.xml [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+                <field name="host"/>
+                <field name="port"/>
+                <field name="user"/>
+                <field name="password"/>
+                <field name="protocol"/>
+            </tree>
+        </field>
+    </record>
+
+    <!-- Branch Form View -->
```
Comment (max3903) [inline connector_equipment/views/backend_equipment.xml]:
```suggestion
    <!-- Backend Form View -->
```

File: connector_equipment/views/maintenance_equipment.xml [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+        <field name="name">maintenance.equipment.connector.page</field>
+        <field name="model">maintenance.equipment</field>
+        <field name="inherit_id" ref="maintenance.hr_equipment_view_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//notebook/page[3]" position="after">
+                <page string="Connector">
+                    <h2>API Information</h2>
+                    <group>
+                        <group>
+                            <field name="backend_equipment_id"/>
```
Comment (max3903) [inline connector_equipment/views/maintenance_equipment.xml]:
```suggestion
                            <field name="backend_id"/>
```

**Review by max3903 — APPROVED** [review]

**Pull request merged**